test/Identity.FunctionalTests/ManagementTests.cs

```diff
@@ -52,5 +56,43 @@ public ManagementTests(ServerFactory serverFactory)
             var email = emails.SentEmails[1];
             await UserStories.ConfirmEmailAsync(email, client);
         }
+
+        [Fact]
+        public async Task CanChangePassword()
+        {
+            // Arrange
+            var principals = new List<ClaimsPrincipal>();
+            var server = ServerFactory.CreateServer(builder =>
+                builder.ConfigureTestServices(s => s.SetupGetUserClaimsPrincipal(user => principals.Add(user))));
+
+            var client = ServerFactory.CreateDefaultClient(server);
+            var newClient = ServerFactory.CreateDefaultClient(server);
+
+            var userName = $"{Guid.NewGuid()}@example.com";
+            var password = $"!Test.Password1$";
+
+            var index = await UserStories.RegisterNewUserAsync(client, userName, password);
+            var manage = await index.ClickManageLinkAsync();
+            var changePassword = await manage.ClickChangePasswordLinkAsync();
+
+            // Act 1
+            var changedPassword = await UserStories.ChangePasswordAsync(changePassword);
+
+            // Assert 1
+            // RefreshSignIn regenerates a new ClaimsPrincipal
+            Assert.NotEqual(GetClaimValue(principals[0]), GetClaimValue(principals[1]));
+
+            // Act 2
+            await UserStories.LoginExistingUserAsync(newClient, userName, $"!Test.Password2$");
+
+            // Assert 2
+            // Signing in again with a different client uses the same ClaimsPrincipal
```

Comment:
nit: The comment is a bit inaccurate. The Security timestamp claim is the only thing that is equal.